packages/fuels-core/src/codec/abi_encoder.rs

```diff
@@ -513,6 +522,48 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn encode_function_with_string_slice_type() -> Result<()> {
```

Comment:
Just to make it clear: right now, we cannot have tests that rely on `forc`-generated bytecode (e.g., a contract using this new `str` slice) because this is still a WIP in the compiler, right? I guess these types of tests would be a future work, once this feature is released in the compiler.

Reply:
That's right. Unfortunately SDK and Sway tests are mutually dependent, so we have to introduce something backwards compatible here first to complete work on the compiler side.

The longer term idea is to deprecate string arrays in favor of string slices anyhow, so the tests will indeed have to be updated here once string slices land on the compiler side.